src/docker.rs

```diff
@@ -48,24 +48,18 @@ fn get_http_connector() -> HttpConnector {
 }
 
 #[cfg(feature = "tls")]
-fn get_docker_for_tcp(tcp_host_str: String) -> Docker {
+fn get_docker_for_tcp(tcp_host_str: String) -> Result<Docker> {
     let http = get_http_connector();
     if let Ok(ref certs) = env::var("DOCKER_CERT_PATH") {
-        // fixme: don't unwrap before you know what's in the box
-        // https://github.com/hyperium/hyper/blob/master/src/net.rs#L427-L428
-        let mut connector = SslConnector::builder(SslMethod::tls()).unwrap();
-        connector.set_cipher_list("DEFAULT").unwrap();
+        let mut connector = SslConnector::builder(SslMethod::tls())?;
+        connector.set_cipher_list("DEFAULT")?;
         let cert = &format!("{}/cert.pem", certs);
         let key = &format!("{}/key.pem", certs);
-        connector
-            .set_certificate_file(&Path::new(cert), SslFiletype::PEM)
-            .unwrap();
-        connector
-            .set_private_key_file(&Path::new(key), SslFiletype::PEM)
-            .unwrap();
+        connector.set_certificate_file(&Path::new(cert), SslFiletype::PEM)?;
```

Comment:
Instead of the generic Ssl error, we could map to a variant `Error::InvalidCertificate`. That would give us something meaningful to say in our `Display` impl without having to print the inner error.

Reply:
SSL error can be returned not just here, it will be generated, for example in docker.rs at line 54: ```let mut connector = SslConnector::builder(SslMethod::tls())?;```, and on line 59: ```connector.set_private_key_file(&Path::new(key), SslFiletype::PEM)?;``` (And there are more).  So we can't really guarantee that all of these errors will be caused by an invalid certificate, and in my opinion, it's better to make just a general SSL error, containing an inner error

Reply:
Yes, we would need more than one variant, not just `InvalidCertificate` but also `InvalidPrivateKey` etc.

Which is why I said "feel free to ignore if you think it becomes too verbose" :)

A general Ssl error is certainly more concise but the error messages will be less helpful because it is not as clear where the error originated.